### PR TITLE
Arith

### PR DESCRIPTION
**Pull request by roconnor-blockstream**
Add various width jets for existing arithmetic operations.

### CODE REVIEW

**Comment by apoelstra** [comment]
`nix-build -A c --argstr wideMultiply int64` fails with this PR.

**Comment by roconnor-blockstream** [comment]
Fixed the `wideMultiply` issue.

**Comment by roconnor-blockstream** [comment]
Oops there are still issues.

**Comment by roconnor-blockstream** [comment]
I think it is good now.

File: Haskell-Generate/GenPrimitive.hs [hunk 1]
```diff
@@ -362,8 +362,10 @@ rawBenchmark "FullSubtract16" = rawBenchmark "FullSubtract32"
 rawBenchmark "FullSubtract64" = rawBenchmark "FullSubtract32"
 rawBenchmark "Multiply8" = rawBenchmark "Multiply32"
 rawBenchmark "Multiply16" = rawBenchmark "Multiply32"
+rawBenchmark "Multiply64" = rawBenchmark "Multiply32"
 rawBenchmark "FullMultiply8" = rawBenchmark "FullMultiply32"
 rawBenchmark "FullMultiply16" = rawBenchmark "FullMultiply32"
+rawBenchmark "FullMultiply64" = rawBenchmark "FullMultiply32"
```
Comment (apoelstra) [inline Haskell-Generate/GenPrimitive.hs:368]: In 5cf9279e61e69f75103ca354ccbdb8c7c62eabf1:

Both of these added lines have 32s at the end which should be 64s.

Found by my local CI running GenPrimitive and confirming consistency between the committed primitiveEnumJet.inc and the one produced by running GenPrimitive.hs :).

Comment (roconnor-blockstream) [reply Haskell-Generate/GenPrimitive.hs:368]: This block of code are stub benchmark timings that need to be filled in later (see the `TODO` above). So we are just reusing existing timings as placeholder.

Comment (apoelstra) [reply Haskell-Generate/GenPrimitive.hs:368]: That's fair enough, but is there some way we can get the output of `GenPrimitive` to match the match the committed file?

The exact diff is
```
57a58
> FULL_MULTIPLY_64,
121a123
> MULTIPLY_64,
```
wherre the new lines are present in the committed file but not the generated one.

Comment (roconnor-blockstream) [reply Haskell-Generate/GenPrimitive.hs:368]: Oh I see.  Yes something seems to have been lost in my rebasing.

Comment (roconnor-blockstream) [reply Haskell-Generate/GenPrimitive.hs:368]: Should be fixed now.

**Review by apoelstra — APPROVED** [review]
ACK 633040cdca4965ffc2e2a64d183d2f0f1b77513c

**Pull request merged**